nupic/research/frameworks/pytorch/model_utils.py

```diff
@@ -552,6 +552,64 @@ def filter_params(
     return filtered_named_params
 
 
+def filter_modules(
+    model,
+    include_modules=None,
+    include_names=None,
+    include_patterns=None,
+):
+    """
+    This iterates through all a model's sub-modules and returns a list of tuples (name,
+    module) for those matches any one of the following conditions
+        1. The module a type is one in 'include_modules'
+        2. The module has a name contained in 'include_names'
+        3. The module's name matches a regex pattern contained in 'include_patterns'
+
+    The regex macthing uses re.match which identifies whether zero or more characters at
+    the beginning of the param name match the regular expression pattern given.
+
+    Example:
+    ```
+    model = resnet50()
+    filter_params(
+        model,
+        include_modules=torch.nn.Linear,  # include linear module
+        include_names=["features.stem"],  # include the stem
+        include_patterns=[".*bn\\d"]  # include modules named `bn` plus a digit
+    )
+    ```
+    """
+
+    include_modules = include_modules or []
+    include_names = include_names or []
+    include_patterns = include_patterns or []
+    assert isinstance(include_names, Collection)
+    assert isinstance(include_patterns, Collection)
+    assert isinstance(include_modules, Collection)
+    include_modules = tuple(include_modules)
+
+    filtered_named_modules = dict()
+    for name, module in model.named_modules():
+
+        # Case 1: The module matches
+        if isinstance(module, include_modules):
+            filtered_named_modules[name] = module
+            continue
+
+        # Case 2: The name is in the white-list.
+        if name in include_names:
+            filtered_named_modules[name] = module
+            continue
```

Comment:
Why if/continue instead of if/elif, the more standard way of doing switch case in python? I don't see the difference.

Reply:
There's no difference. I think I wrote it that way for readability. I wanted to make it clear that each case is distinct, but case 3 follows a different pattern of checking, one involving a `for` loop.